packages/common/src/core/types/Metrics.ts

```diff
@@ -0,0 +1,171 @@
+import { IQuery } from "../../search/types/IHubCatalog";
+import { ServiceAggregation } from "./DynamicValues";
+import { IReference } from "./IReference";
+
+/**
+ * Defines the information required from a Metric
+ */
+export interface IMetric {
+  /**
+   * Identifier for the metric. This must be static over time or linkages
+   * between items will break. Can be the camelCase version of the name
+   * as long as it's a valid Javascript property name (can't start with numbers).
+   */
+  id: string;
```

Comment:
Jupe and I were thinking about this yesterday when you gave your proposal for moving this to a dynamic field for the value:

I understand the concern, but I had imagined the metric id would be the cardId when used in a site's layout -- so it very well could start with numbers in that case...there's no way to safeguard this by wrapping it with quotes when we create the property or refer to it, right?

If this is the camelCase version of the name, I guess this same safeguarding comes into play with naming on the project side of things too -- so that would just be a note for front-end implementation, that project metrics wouldn't be able to start with numbers.

Reply:
We can also use `createId(..)`, which takes an alpha char to create a random id. It's just easier to debug if it's something readable vs `c1238`. Main thing is that it can't change over time. So if it's not set, create it from camelCased name, removing leading numbers. And never change it :)